src/terminal/adapter/adaptDispatch.cpp

```diff
@@ -1469,7 +1541,10 @@ bool AdaptDispatch::ForwardTab(const size_t numTabs)
         }
     }
 
-    _pConApi->SetCursorPosition({ column, row });
+    cursor.SetXPosition(column);
+    cursor.SetDelay(false);
+    cursor.SetIsOn(true);
+    cursor.SetHasMoved(true);
```

Comment:
Ah okay I found it. These come from: `SCREEN_INFORMATION::SetCursorPosition` 

https://github.com/microsoft/terminal/blob/0651d92dba09aaf13a807f1502a10bef159e4f5d/src/host/screenInfo.cpp#L1728-L1740

`IsOn` is always passed as true in `ApiRoutines::SetConsoleCursorPositionImpl`, so this right here is assuming that the console has focus.... that's probably okay. I don't entirely know the origin of that HAS_FOCUS check.

This may be served well by having a helper - I think I'm gonna have to introduce focus tracking even for conpty fairly soon.

Reply:
My impression was the the `HAS_FOCUS` check was just an optimization, and it shouldn't harm to set these flags anyway. The `TimerRoutine` that processes this stuff has it's own `HAS_FOCUS` check that prevents it triggering a notification if the window isn't focused.

https://github.com/microsoft/terminal/blob/ce6ce50e7e9eeb2d7377527d9512da9d43674bc0/src/host/CursorBlinker.cpp#L86-L89

If anything, I think the existing `HAS_FOCUS` check might cause it lose some move events, and could be a bug. But again I thought this stuff should probably be investigated further in a follow up issue. I didn't want to get too bogged down trying to resolve side issues like this if I could avoid it. As long as we aren't making thing worse in the meantime.